Debug initialization of gossip network

When writing acceptance tests for block gossiping, I had to add a call to Thread.sleep() in order for the gossiping to work.  Otherwise, attempts to gossip would fail because the gossip layer didn't have any peers available for gossiping.  We should investigate why this is required and see if there's a way to remove the extra wait: https://github.com/PegaSysEng/artemis/pull/986/files#diff-66d1036b3a68c67245aaaaa19fb588efR74

Output when the wait is removed:

```
2019-11-06 17:28:03.721-05:00 | P2PService-event-thread-0 | TRACE  | GossipTopicHandler | Failed to gossip message on Topic(topic=/eth2/beacon_block/ssz)
java.util.concurrent.CompletionException: io.libp2p.etc.types.NothingToCompleteException
	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:367) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:376) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1074) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: io.libp2p.etc.types.NothingToCompleteException
	at io.libp2p.etc.types.AsyncExtKt.anyComplete(AsyncExt.kt:62) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.etc.types.AsyncExtKt.anyComplete(AsyncExt.kt:59) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.pubsub.gossip.GossipRouter.broadcastOutbound(GossipRouter.kt:123) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.pubsub.AbstractRouter$publish$1.invoke(AbstractRouter.kt:48) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.pubsub.AbstractRouter$publish$1.invoke(AbstractRouter.kt:24) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.etc.types.AsyncExtKt$submitAsync$1.get(AsyncExt.kt:51) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at io.libp2p.etc.types.AsyncExtKt$submitAsync$1.get(AsyncExt.kt) ~[jvm-libp2p-minimal-0.1.0-RELEASE.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) ~[?:?]
	... 6 more
```

--- a/networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/GossipMessageHandlerIntegrationTest.java
+++ b/networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/GossipMessageHandlerIntegrationTest.java
@@ -81,7 +81,7 @@ public class GossipMessageHandlerIntegrationTest {
           assertThat(node2.network().getPeerCount()).isEqualTo(2);
           assertThat(node3.network().getPeerCount()).isEqualTo(1);
         });
-    // TODO (#1855): debug this - we shouldn't have to wait here
+    // Wait for subscriptions to complete (jvm-libp2p does this asynchronously)
     Thread.sleep(2000);
 
     // Propagate block from network 1
@@ -131,7 +131,7 @@ public class GossipMessageHandlerIntegrationTest {
           assertThat(node3.network().getPeerCount()).isEqualTo(1);
         });
 
-    // TODO (#1855): debug this - we shouldn't have to wait here
+    // Wait for subscriptions to complete (jvm-libp2p does this asynchronously)
     Thread.sleep(2000);
 
     // Propagate invalid block from network 1

--- a/sync/src/integration-test/java/tech/pegasys/teku/sync/BlockPropagationIntegrationTest.java
+++ b/sync/src/integration-test/java/tech/pegasys/teku/sync/BlockPropagationIntegrationTest.java
@@ -93,7 +93,7 @@ public class BlockPropagationIntegrationTest {
           assertThat(node1.network().getPeerCount()).isEqualTo(1);
           assertThat(node2.network().getPeerCount()).isEqualTo(1);
         });
-    // TODO (#1855): debug this - we shouldn't have to wait here
+    // Wait for subscriptions to complete (jvm-libp2p does this asynchronously)
     Thread.sleep(2000);
 
     // Update slot so that blocks can be imported

